pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -0,0 +1,3239 @@
+
+    #: Stimulus Response
+    SpectrumAnalyzer = "SA"
+
+    #: Spectrum Analyeze
+    StimulusResponse = "SR"
+
+
+class SweepOut(StrEnum):
+    """Enumeration."""
+
+    #: 0 - 10V Ramp
+    Ramp = "RAMP"
+
+    #: DC Ramp 0.5V / GHz
+    Fav = "FAV"
+
+
+class MixerMode(StrEnum):
+    """Enumeration to represent the Mixer Mode of the HP8561B."""
+
+    #: Mixer Mode Internal
+    Internal = "INT"
+
+    #: Mixer Mode External
+    External = "EXT"
+
+
+class SourceLevelingControlMode(StrEnum):
+    """Enumeration to represent the Source Leveling Control Mode of the
+    HP8560A."""
+
+    #: Source Leveling Control Mode Internal
+    Internal = "INT"
+
+    #: Source Leveling Control Mode External
+    External = "EXT"
+
+
+class PeakSearchMode(StrEnum):
+    """Enumeration to represent the Marker Peak Search Mode."""
+
+    #: Place marker to the highest value on the trace
+    High = "HI"
+
+    #: Place marker to the next highest value on the trace
+    NextHigh = "NH"
+
+    #: Place marker to the next peak to the right
+    NextRight = "NR"
+
+    #: Place marker to the next peak to the left
+    NextLeft = "NL"
+
+
+class CouplingMode(StrEnum):
+    """Enumeration to represent the Coupling Mode."""
+
+    #: AC
+    AC = "AC"
+
+    #: DC
+    DC = "DC"
+
+
+class DemodulationMode(StrEnum):
+    """Enumeration to represent the Demodulation Mode."""
+
+    #: Amplitude Modulation
+    Amplitude = "AM"
+
+    #: Frequency Modulation
+    Frequency = "FM"
+
+    #: Demodulation Off
+    Off = "OFF"
+
+
+class TriggerMode(StrEnum):
+    """Enumeration to represent the different trigger modes"""
+
+    #: External Mode
+    External = "EXT"
+
+    #: Free Running
+    Free = "FREE"
+
+    #: Line Mode
+    Line = "LINE"
+
+    #: Video Mode
+    Video = "VID"
+
+
+class TraceDataFormat(StrEnum):
+    """Enumeration to represent the different trace data formats."""
+
+    #: A-Block format
+    A_BLOCK = "A"
+
+    #: Binary format
+    BINARY = "B"
+
+    #: I-Block format
+    I_BLOCK = "I"
+
+    #: ASCII format
+    ASCII = "M"
+
+    #: Real numbers format like are in Hz, volts, watts, dBm, dBmV, dBuV, dBV, or seconds.
+    REAL = "P"
+
+
+class FrequencyReference(StrEnum):
+    """Enumeration to represent the frequency reference source."""
+
+    #: Internal Frequency Reference
+    Internal = "INT"
+
+    #: External Frequency Standard
+    External = "EXT"
+
+
+class DetectionModes(StrEnum):
+    """Enumeration to represent the Detection Modes."""
+
+    #: Negative Peak Detection
+    NegativePeak = "NEG"
+
+    #: Normal Peak Detection
+    Normal = "NRM"
+
+    #: Positive Peak Detection
+    PositivePeak = "POS"
+
+    #: Sampl Mode Detection
+    Sample = "SMP"
+
+
+class AmplitudeUnits(StrEnum):
+    """Enumeration to represent the amplitude units."""
+
+    #: DB over millit Watt
+    DBM = "DBM"
+
+    #: DB over milli Volt
+    DBMV = "DBMV"
+
+    #: DB over micro Volt
+    DBUV = "DBUV"
+
+    #: Volts
+    V = "V"
```

Comment:
The way I right now have implemented / used the enumerations, it wouldn't easily work.
The instrument responds with upper case, but the value maps would be mixed case.
I tried shortly implementing a case insensitive StrEnum version, but this would not work with all current supported python versions.

Do you think this change is worth reworking all the attributes get_- ways  ?

Reply:
First, The enum name can be all caps (as enums are that normally), but you can change it, if you like and it is possible. Second, the enum string can be all caps as well, as it is used by the instrument.

You should adjust the comments though (which do not influence the code).


One note: you can "call" an enum, and then it uses the string value (`some_enum("string value")`). And you can use the Index access `somd_enum["enum_name"] `.

Maybe that helps.